src/main.cpp

```diff
@@ -4067,9 +4074,13 @@ bool CheckIndexAgainstCheckpoint(const CBlockIndex *pindexPrev,
     return true;
 }
 
-bool ContextualCheckBlockHeader(const CBlockHeader &block, CValidationState &state, CBlockIndex *const pindexPrev)
+bool ContextualCheckBlockHeader(const CBlockHeader &block, CValidationState &state, CBlockIndex *const pindexPrev, bool *pWeak)
 {
     const Consensus::Params &consensusParams = Params().GetConsensus();
+
+    *pWeak = (CheckProofOfWork(block.GetHash(), WeakBlockProofOfWork(block.nBits), Params().GetConsensus()) &&
```

Comment:
nit: might be worth breaking these two conditions into named variables.

Reply:
Done (part of a couple yet to be published commits)